app/src/main/java/org/greenstand/android/TreeTracker/database/dao/PlanterDao.kt

```diff
@@ -21,6 +21,10 @@ interface PlanterDao {
     @Query("SELECT * FROM planter_identifications WHERE photo_url IS NULL")
     fun getPlanterIdentificationsToUpload(): List<PlanterIdentificationsEntity>
 
+    @Transaction
+    @Query("SELECT * FROM planter_identifications WHERE identifier = :identifier")
+    fun getPlanterIDByIdentifier(identifier: String?): Int
```

Comment:
I don't know if you resolved this issue already, but this query `SELECT *` will return All fields so you can change the return type to `PlanterIdentificationsEntity` OR change the query to `select planterDetailsId` and it will match the return type

Reply:
Thanks, @mozarty , I am updating now the code with the latest changes to use the TreeManager instead this way I've done here. 
I'll come back with the update and then if you have some time to review my code it will be great!